mobile/native/src/orderbook.rs

```diff
@@ -71,6 +78,10 @@ pub fn subscribe(secret_key: SecretKey, runtime: &Runtime) -> Result<()> {
             let mut stream =
                 spawn_blocking(move || orderbook_client::subscribe_with_authentication(url, authenticate)).await.expect("joined task not to panic");
 
+            if let Err(e) =
+                orderbook_status.send(ServiceStatus::Online) {
+                    tracing::warn!("Cannot update orderbook status: {e:#}");
+                };
```

Comment:
🤔I'm not sure that passing anything but that dedicated watch channel would be a good idea, this code ultimately
I could encapsulate it in a different API to swallow the errors, but we're not doing this anywhere else.

At the end of the day, we could might as well write `expect` as it should not fail, the receiver will not be dropped.

if we want to encapsulate the boilerplate, we could restore our old `log_error` (with a new `log_warn`) macros to avoid this annoying construct! 
thoughts?

Reply:
I mean, it already knows about the module because you have to manually set the status to `health::ServiceStatus::Online`.

My proposal was to unburden this module from having to choose types and just give it a method to call. The fact that we are using an enum for the states or a `tokio::sync::watch::Sender` is a bit of an implementation detail.

The error handling is annoying, but we don't have to encapsulate it in the same way. I'm not a huge fan of macros these days, but if you are very keen I think it would make sense to use it here.